scripts/delete_resources.sh

```diff
@@ -1,149 +1,240 @@
-
-
-  #Delete Route53 record
-  if [[ "$R53_RECORD" == "${NAME_PREFIX_LOWER}."* ]]; then
-    echo "Deleting Route53 record ${R53_RECORD}"
-    CHANGE_STATUS=$(aws route53 change-resource-record-sets --hosted-zone-id "${R53_ZONE_ID}" \
-      --change-batch '{"Changes": [
-              {
-                  "Action": "DELETE",
-                  "ResourceRecordSet": {
-                      "Name": "'"${R53_RECORD}"'",
-                      "Type": "CNAME",
-                      "TTL": 300,
-                      "ResourceRecords": [
-                          {
-                              "Value": "'"${RECORD_VALUE}"'"
-                          }
-                      ]
-                  }
-              }
-          ]
-      }')
-    STATUS_ID=$(echo "$CHANGE_STATUS" | jq -r '.ChangeInfo.Id')
-    #Get status from the change
-    aws route53 wait resource-record-sets-changed --id "$STATUS_ID"
-    echo "Successfully deleted Route53 record ${R53_RECORD}: status: ${STATUS_ID}"
+    --query "ResourceRecordSets[?starts_with(Name, '${NAME_PREFIX_LOWER}')].ResourceRecords[0].Value" --output text)
+  echo "RECORD_VALUE:
+  $RECORD_VALUE"
+  RECORD_COUNT=$(echo $RECORD_VALUE | xargs -n1 echo | wc -l)
+  echo "Total Record COUNT is: $RECORD_COUNT"
+  if [ "${RECORD_COUNT}" == 0 ]; then
+    echo "No Route53 records found for prefix $1. Nothing to delete."
   else
-    echo "No Route53 record found"
+    for i in $(echo $RECORD_VALUE | xargs -n1 echo)
+    do 
+        NAME="$(echo "$i" | cut -d "-" -f1)-r53.qa.rancher.space."
+        VALUE=$i
+        echo " NAME: $NAME"
+        echo " VALUE: $VALUE"
+        echo "
+{\"Changes\": [
+        {
+            \"Action\": \"DELETE\",
+            \"ResourceRecordSet\": {
+                \"Name\": \"${NAME}\",
+                \"Type\": \"CNAME\",
+                \"TTL\": 300,
+                \"ResourceRecords\": [
+                    {
+                        \"Value\": \"${VALUE}\"
+                    }
+                ]
+            }
+        }
+    ]
+}" > payload.json
+      cat payload.json
+      echo "Deleting Route53 record ${NAME} for prefix $1"
+      CHANGE_STATUS=$(aws route53 change-resource-record-sets --hosted-zone-id "${R53_ZONE_ID}" --change-batch file://"${PWD}"/payload.json)
+      echo "CHANGE STATUS: 
+      $CHANGE_STATUS"
+      STATUS_ID=$(echo "$CHANGE_STATUS" | grep Id: | cut -d " " -f 4)
+      echo "Status ID: $STATUS_ID"
+      #Get status from the change
+      aws route53 wait resource-record-sets-changed --id "$STATUS_ID"
+      echo "Successfully deleted Route53 record ${NAME}: status: ${STATUS_ID}"
+      exit
+    done
+    rm -rf "${PWD}"/payload.json
   fi
+}
+
+delete_all_resources () {
+  delete_ec2_instances "$1"
+  delete_db_resources "$1"
+  delete_lb_resources "$1"
+  delete_target_groups "$1"
+  delete_route_s3 "$1"
+}
+
+
+while getopts r:h OPTION
+do 
+  case "${OPTION}"
+    in
+    r) RESOURCES=${OPTARG};;
```

Comment:
jenkins jobs get created only with lower case values. AFAIK the jobs will fail if we had upper case or special characters(i could be wrong). But if they had upper case when they manually created a setup, changing to lower case would prevent them from using this to delete that setup. i feel putting that restriction should not be on this script. the user should know what they want to delete and provide that.

Reply:
the worst that will happen, is we wont find resources for that name and we wont do anything. but in the case of having resource names with capitals - it will still look for it and delete them if we want to. there is value in that.